<launch>

  <!-- https://github.com/jsk-ros-pkg/jsk_pr2eus/blob/master/pr2eus_tutorials/launch/pr2_tabletop.launch とhttps://github.com/jsk-ros-pkg/jsk_pr2eus/blob/master/pr2eus_tutorials/launch/gazebo/pr2_tabletop_sim.launch -->
  <arg name="run_rviz" default="true" />
  <arg name="publish_objects_tf" default="true" />
  <arg name="launch_tracking" default="false" />
  <arg name="manager" default="pr2_tabletop_object_detector_nodelet_manager"/>
  <arg name="machine" default="localhost" />
  <arg name="load_machine" default="true" />

  <include file="$(find pr2_machine)/$(env ROBOT).machine" unless="$(arg load_machine)" />

  <include file="$(find jsk_pcl_ros)/sample/tabletop_object_detector.launch">
    <arg name="input" value="/kinect_head/depth_registered/throttled/points" />
    <arg name="sensor_frame" value="/head_mount_kinect_rgb_optical_frame" />
    <arg name="manager" value="$(arg manager)" />
    <arg name="machine" value="$(arg machine)" />
    <arg name="launch_openni" value="false" />
    <arg name="launch_tracking" value="$(arg launch_tracking)" />
    <arg name="launch_rviz" value="false" />
    <arg name="publish_tf" value="$(arg publish_objects_tf)" />
  </include>

  <node if="$(arg run_rviz)"
        pkg="rviz" name="pr2_rviz" type="rviz"
        args="-d $(find pr2eus_tutorials)/config/pr2_tabletop.rviz" />
</launch>
